src/components/MaterialUI/Inputs/DatePicker.js

```diff
@@ -133,10 +137,25 @@ const WrappedDatePicker = ({
 	const classes = useStyles({ readOnly });
 	const startDate = value ? new Date(value) : null;
 	const disabledCls = classNames({ [classes.disabled]: props.disabled });
-
+	const localizedTimeZoneName = useSelector(namedLookupLocalizedSelector("customer", "TimeZone", timePickerTimeZone));
+	const setLocalZone = (date, timezone) => {
+		const dateWithoutZone = moment.tz(date, timezone).format("YYYY-MM-DDTHH:mm:ss.SSS");
+		const localZone = moment(dateWithoutZone).format("Z");
+		const dateWithLocalZone = [dateWithoutZone, localZone].join("");
+		return new Date(dateWithLocalZone);
+	}
+	  
+	const setOtherZone = (date, timezone) => {
+		console.log(timezone);
+		const dateWithoutZone = moment(date).format("YYYY-MM-DDTHH:mm:ss.SSS");
+		const otherZone = moment.tz(date, timezone).format("Z");
+		const dateWithOtherZone = [dateWithoutZone, otherZone].join("");
+		return new Date(dateWithOtherZone);
+	}
+	
 	const updateDate = (date, metadata) => {
 		if (onChange) {
-			onChange(date, metadata);
+			onChange(date ? setOtherZone(date, getTimeZoneByName(timePickerTimeZone)) : null, metadata)
 		}
```

Comment:
overwrite local timezone just when custom timezone is specified and time is used for picker, not always